packages/ifpack2/example/CMakeLists.txt

```diff
@@ -21,11 +21,73 @@ TRIBITS_ADD_EXECUTABLE(
 )
 
 
- TRIBITS_ADD_TEST(
-    BlockTriDiagonalSolver
-    NAME BlockTriDiExample
-    ARGS ""
-    COMM serial mpi
-    NUM_MPI_PROCS 1-4
-    STANDARD_PASS_OUTPUT
-    )
+TRIBITS_ADD_TEST(
+  BlockTriDiagonalSolver
+  NAME BlockTriDiExample
+  ARGS ""
+  COMM serial mpi
+  NUM_MPI_PROCS 1-4
+  STANDARD_PASS_OUTPUT
+)
+
+ASSERT_DEFINED (
+  ${PACKAGE_NAME}_ENABLE_Xpetra
+  ${PACKAGE_NAME}_ENABLE_Galeri
+)
+
+IF(${PACKAGE_NAME}_ENABLE_Xpetra AND ${PACKAGE_NAME}_ENABLE_Galeri)
+
+  set(COMMON_ARGS "--withoutOverlapCommAndComp --matrixType=Laplace3D --blockSize=11 --nx=256 --ny=128 --nz=128 --tol=0 --numIters=5 --withStackedTimer --numRepeats=5")
+  set(BTDS_ARGS "${COMMON_ARGS} --sublinesPerLine=1 --sublinesPerLineSchur=1")
+  set(BTDS_SCHUR_ARGS "${COMMON_ARGS} --sublinesPerLine=1 --sublinesPerLineSchur=2")
+  set(BTDS_DEFAULT_SCHUR_ARGS "${COMMON_ARGS} --sublinesPerLine=1 --sublinesPerLineSchur=-1")
+  set(BLOCK_JACOBI_ARGS "${COMMON_ARGS} --sublinesPerLine=-1")
+
+  set(TEST_ARGS ${BTDS_ARGS} ${BTDS_SCHUR_ARGS} ${BTDS_DEFAULT_SCHUR_ARGS} ${BLOCK_JACOBI_ARGS})
+  set(TEST_NAMES "BTDS" "BTDS_Schur" "BTDS_Schur_Default" "BTDS_Jacobi")
+  set(ARCHS "SERIAL" "CUDA" "HIP")
+
+  list(LENGTH TEST_ARGS len1)
+  math(EXPR N_TESTS "${len1} - 1")
+
+  ASSERT_DEFINED(
+    Tpetra_INST_SERIAL
+    Tpetra_INST_CUDA
+    Tpetra_INST_HIP
+  )
+
+  foreach(ARCH ${ARCHS})
+    IF (Tpetra_INST_${ARCH})
+      MESSAGE(STATUS "Ifpack2: BlockTriDiagonalSolver ${ARCH} test ENABLED")
+      set(COUNTER 2)
+      set(MAX_COUNT 8)
+
+      set(NP 1)
+      set(COUNTER_INIT 0)
+      WHILE(COUNTER_INIT LESS COUNTER)
+        math(EXPR NP "${NP} * 2")
+        math(EXPR COUNTER_INIT "${COUNTER_INIT} + 1")
+      ENDWHILE()
+
+      WHILE(COUNTER LESS MAX_COUNT)
+        foreach(I_TEST RANGE ${N_TESTS})
+          list(GET TEST_ARGS ${I_TEST} TEST_ARG)
+          list(GET TEST_NAMES ${I_TEST} TEST_NAME)      
+          TRIBITS_ADD_TEST(
```

Comment:
Same issue as with PerformanceCGSolve I think; the Watchr filenames will conflict across backends and test cases, if the goal is to generate Watchr files. See BlockTriDi.cpp, line 686 for where this name gets constructed. You could create unique names based on backend, plus the ``problemName`` command line argument (maybe something like ``--problemName=Galeri_${TEST_NAME}``).

Reply:
Does this not result in tests names "${TEST_NAME}_${ARCH}_${NUM_MPI_PROCS}"

Reply:
Yes:

```
Test project /home/knliege/dev/trilinos_all/Trilinos_Weaver_B
    Start 5: Ifpack2_BTDS_CUDA_MPI_4
1/4 Test #5: Ifpack2_BTDS_CUDA_MPI_4 .................   Passed    8.08 sec
    Start 6: Ifpack2_BTDS_Schur_CUDA_MPI_4
2/4 Test #6: Ifpack2_BTDS_Schur_CUDA_MPI_4 ...........   Passed    8.41 sec
    Start 7: Ifpack2_BTDS_Schur_Default_CUDA_MPI_4
3/4 Test #7: Ifpack2_BTDS_Schur_Default_CUDA_MPI_4 ...   Passed    7.99 sec
    Start 8: Ifpack2_BTDS_Jacobi_CUDA_MPI_4
4/4 Test #8: Ifpack2_BTDS_Jacobi_CUDA_MPI_4 ..........   Passed    7.78 sec

100% tests passed, 0 tests failed out of 4

Subproject Time Summary:
Ifpack2    = 129.04 sec*proc (4 tests)

Total Test time (real) =  32.36 sec
bash-4.4$ 
```

Reply:
Sure but the watchr files are not named based on the ctest name (maybe they should incorporate this, but the original plan was that you can pass flags to give them problem-specific names like "SPARC Single GPU".

Reply:
We would need to modify these too then https://github.com/trilinos/Trilinos/blob/bf922e75428ef7dd925811f0a532d05dd9cff808/packages/muelu/test/scaling/CMakeLists.txt#L741-L811

Reply:
@kliegeois You're right, for example on hops this test might be running on both serial and cuda, and if that's the case one set of files is overwriting the other.

Reply:
Turns out this was OK on hops because Cuda was the only Tpetra backend enabled, but it would be a problem if multiple backends were ever enabled